contracts/FindMarketAuctionEscrow.cdc

```diff
@@ -272,34 +274,59 @@ pub contract FindMarketAuctionEscrow {
 			emit ForAuction(tenant:self.getTenant().name, id: id, seller:seller, sellerName: FIND.reverseLookup(seller), amount: balance, auctionReservePrice: saleItem.auctionReservePrice,  status: status, vaultType:saleItem.vaultType.identifier, nft: nftInfo,  buyer: buyer, buyerName: buyerName, endsAt: saleItem.auctionEndsAt)
 		}
 
-		pub fun getItemForSaleInformation(_ id:UInt64) : FindMarket.SaleItemInformation {
+		pub fun getItemForSaleInformation(_ id:UInt64) : FindMarket.SaleItemInformation? {
 			pre {
 				self.items.containsKey(id) : "Invalid id=".concat(id.toString())
 			}
-			return FindMarket.SaleItemInformation(self.borrow(id))
 
+			let item=self.borrow(id)
+			if item.pointer.valid() {
+				return FindMarket.SaleItemInformation(self.borrow(id))
+			}
+			return nil
 		}
 
 		pub fun getItemsForSale(): [FindMarket.SaleItemInformation] {
```

Comment:
I would love to find a way to merge this and ghost listing so we do not have to iterate twice. Just return a stuct that has two arrays and we can then merge it. Also make an interface for the SaleItem Collections in FindMarket that they all follow so we can make the code in *status.cdc better